### PR TITLE
Allow disabling keep-alive

### LINKED ISSUE #14538: fetch does not respect `keepalive: false`
### What version of Bun is running?

1.1.30+7996d06b8

### What platform is your computer?

Darwin 24.0.0 arm64 arm

### What steps can reproduce the bug?

I have tried the following 2 ways to set `Connection` header to `close`:

1. By etting `keepalive` to `false`:
```js
await fetch('http://localhost:3000', {
  verbose: true,
  keepalive: false,
});
```
2. By setting `Connection` header to false manually:
```js
await fetch('http://localhost:3000', {
  headers: {
    Connection: 'close',
  },
  verbose: true,
});
```

### What is the expected behavior?

In the debug log, `Connection` header should be `close`

### What do you see instead?

`Connection` header is set to `keep-alive` in both cases:
```
[fetch] > HTTP/1.1 GET http://localhost:3000/
[fetch] > Connection: keep-alive
[fetch] > User-Agent: Bun/1.1.30
[fetch] > Accept: */*
[fetch] > Host: localhost:3000
[fetch] > Accept-Encoding: gzip, deflate, br

[fetch] < 200 OK
[fetch] < content-type: text/plain;charset=utf-8
[fetch] < Date: Sun, 13 Oct 2024 22:16:15 GMT
[fetch] < Content-Length: 11
```

### What
Bun's implementation of `fetch` does not respect the `keep-alive` option or `Connection` header override.

### Context
I know that per Bun's [documentation and recommendation](https://bun.sh/docs/api/fetch#connection-pooling-http-keep-alive), connection-pooling is recommended. However, there are APIs (out of my control) that my application talks to that rejects the call if `keep-alive` is set to `true`. In addition, [WHATWG](https://fetch.spec.whatwg.org/#request-keepalive-flag) states that `keepalive` should by default be `false`.

This behavior in Bun's `fetch` prevents me from moving to it from `node:http`.

### Request
Bun's `fetch` should set `keep-alive` to false by default, or at least provide a way to turn disable `keep-alive`. 

### References
1. See [a similar issue raised to nodejs/undici](https://github.com/nodejs/undici/issues/2169)
2. [WHATWG spec](https://fetch.spec.whatwg.org/#request-keepalive-flag) for `fetch`

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

Fixes #14538

### How did you verify your code works?

Test

### CODE REVIEW

**Comment by robobun** [comment]
@Jarred-Sumner, your commit 54d1cd24c02ef21e7bd8b99ea30d44cf07c78c17 is building: [`#4814`](https://buildkite.com/bun/bun/builds/4814)

File: test/js/web/fetch/fetch-keepalive.test.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    async fetch(req) {
+      return new Response(JSON.stringify(req.headers.toJSON()));
+    },
+  });
+  {
+    const res = await fetch(`http://localhost:${server.port}`, {
+      keepalive: false,
+    });
+    const headers = await res.json();
+    console.log(headers);
```
Comment (cirospaciari) [inline test/js/web/fetch/fetch-keepalive.test.ts]:
```suggestion
```

File: test/js/web/fetch/fetch-keepalive.test.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    expect(headers.connection).toBeUndefined();
+  }
+
+  {
+    const res = await fetch(`http://localhost:${server.port}`, {
+      keepalive: true,
+    });
+    const headers = await res.json();
+    console.log(headers);
```
Comment (cirospaciari) [inline test/js/web/fetch/fetch-keepalive.test.ts]:
```suggestion
```

File: test/js/web/fetch/fetch-keepalive.test.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  {
+    const res = await fetch(`http://localhost:${server.port}`, {
+      keepalive: false,
+      headers: {
+        "Connection": "HELLO!",
+      },
+    });
+    const headers = await res.json();
+    console.log(headers);
```
Comment (cirospaciari) [inline test/js/web/fetch/fetch-keepalive.test.ts]:
```suggestion

```

**Review by cirospaciari — APPROVED** [review]

**Pull request merged**